x-pack/docs/en/security/securing-communications/update-tls-certificates.asciidoc

```diff
@@ -0,0 +1,529 @@
+[[update-node-certs]]
+== Updating node security certificates
+If your cluster certificates expired or you have a new Certificate Authority
+(CA), then you'll need to update the security certificates for both the
+transport and HTTP layers. When updating your node security certificates, there
+are two main scenarios that you'll encounter:
+
+* <<update-node-certs-same,Your certificate authority (CA) or key remains the same>>
+* <<update-node-certs-different,Your CA certificate or key changes>>
+
+Regardless of the scenario, {es} monitors the SSL resources for updates
+on a five-second interval. You can just copy the new certificate and key files
+(or keystore) into the {es} configuration directory and your nodes will detect
+the changes. 
+
+Because {es} doesn't reload the `elasticsearch.yml` configuration,
+you must use *the same file names* so that {es} refreshes the security
+certificates and keys. This method of updating certificates is preferred
+because it doesn't require a full cluster restart.
+
+If you need to update the `elasticsearch.yml` configuration or change
+certificate passwords that are stored in the
+<<secure-settings,secure settings>>, then you must complete a full cluster
+restart. {es} will not reload changes for passwords stored in the secure
+settings. We recommend <<use-rolling-restarts,rolling restarts>> when restarting your cluster.
+
+[[use-rolling-restarts]]
+.Rolling restarts are preferred
+****
+While it's possible to do an in-place update for security certificates, using
+a <<restart-cluster-rolling,rolling restart>> on your cluster is safer. An in-place update avoids some
+complications of a rolling restart, but incurs the following risks:
+
+* If you use PEM files, your certificate and key are in separate files. You
+must update both files _simultaneously_ or the node might experience a temporary
+period where it cannot establish new connections.
+* Updating the certificate and key does not automatically force existing
+connections to refresh. This means that even if you make a mistake, a node can
+seem like it's functioning but only because it still has existing connections.
+It's possible that a node will be unable to connect with other nodes, rendering
+it unable to recover from a network outage or node restart.
+****
+
+[[update-node-certs-same]]
+=== Update security certificates with the same CA
+++++
+<titleabbrev>With the same CA</titleabbrev>
+++++
+
+If your CA hasn't changed, you can just generate a new certificate, save it with
```

Comment:
2 lines ago we said that in place is not the preferred way to go and we're suggesting now that users should do that. Maybe also always qualify "certificate" with either "node.. " or "CA.. " so that we don't confuse readers?

Reply:
But a rolling restart is only required if the keystore is password protected, the password is stored in the {es} keystore, _and_ the password needs to change, correct? Or do we recommend that users perform a rolling restart regardless?